src/modsim2/similarity/mmd.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+from sklearn import metrics
+
+from modsim2.similarity.embeddings import EMBEDDING_FN_DICT
+
+MMD_KERNEL_DICT = {
+    "rbf": metrics.pairwise.rbf_kernel,
+    "laplace": metrics.pairwise.laplacian_kernel,
+}
+
+
+def mmd(
+    array_A: np.ndarray, array_B: np.ndarray, embedding_name: str, kernel_name: str
+) -> float:
+    """
+    Computes maximum mean discrepancy (MMD) for A and B. Given a kernel
+    embedding k(X,X'), computes
+
+        MMD^2 = 1/(N_A^2)*sum(k(A,A)) + 1/(N_B^2)*sum(k(B,B)) -
+                    2/(N_A * N_B)*sum(k(A,B))
+
+    Note that A and B must be matrices for the MMD. Currently mmd()
+    will reshape arbitarily size np.ndarrays to matrices. However,
+    other options include feature embeddings which will be explored in
+    the future.
+
+    Args:
+        array_A: The first image dataset
+        array_B: The second image dataset
+        kernel_name: An appropriate kernel embedding. See kernel_dict
```

Comment:
missing embedding_name here

Reply:
I've added this

And yes - given the way MMD works and the fact that we stratify our drops by class, I don't think dropping should change this (or probably a lot of the other metrics too - though PAD might be vulnerable to it)